pkg/engine/handlers/mutation/load_targets.go

```diff
@@ -57,31 +58,32 @@ func loadTargets(ctx context.Context, client engineapi.Client, targets []kyverno
 }
 
 func resolveSpec(i int, target kyvernov1.TargetResourceSpec, ctx engineapi.PolicyContext, logger logr.Logger) (kyvernov1.TargetSelector, error) {
-	kind, err := variables.SubstituteAll(logger, ctx.JSONContext(), target.Kind)
+	var s kyvernov1.TargetSelector
+	jsonData, err := json.Marshal(target.TargetSelector)
```

Comment:
How do we resolve variables in `targets.preconditions`? Have you tested such use cases?

Reply:
Since the mutate test passed after we made this change, and in the original code we replaced variables for name, namespace, api version and kind we can conclude that the reason the failure happened is because we are replacing variables that were never supposed to be replaced. and so we are once again passing only the relevant bit to the variables replacer

Since now the tests passes also then its safe to assume that another part of the code is handling this preconditions part. I am yet to discover what it is

But the point is, in the context of this PR the change is now valid and passing

Reply:
Ok it's handled here: https://github.com/kyverno/kyverno/blob/354d424e284e5c746f89199654f549a5131da27c/pkg/engine/handlers/mutation/mutate_existing.go#L84-L89